gamification/models.py

```diff
@@ -0,0 +1,206 @@
+                       for performing an activity
+        :param activity_string: represents the activity
+                                performed by the participant
+        :param performed_at: a datetime object representing the date and time
+                             when this activity was performed
+        :param updated_at: a datetime object representing the date and time when
+                           this activity was performed
+        """
+        self.update_score_and_level(points)
+        self.add_activity(points, activity_string, performed_at, updated_at)
+
+    def deduct_points(self, points_to_deduct, activity_string,
+                      performed_at, updated_at):
+        """
+        Deduct points for performing some specific activities.
+        """
+        self.add_points(points_to_deduct, activity_string, performed_at, updated_at)
+
+    def find_level_for_score(self, score):
+        """
+        Find suitable level based on the total score earned.
+        """
+        level = Level.objects.get(min_score__lte=score, max_score__gt=score)
+        return level
+
+    def update_score_and_level(self, points):
+        """
+        Update score and level based on points.
+        """
+        if points < 0 and self.score < abs(points):
+            new_score = self.score = 0
+        else:
+            self.score += points
+            new_score = self.score
+
+        new_level = self.find_level_for_score(new_score)
+        if new_level.number > self.level.number:
+            self.level = new_level
+
+    def add_activity(self, points, activity_string, performed_at, updated_at):
+        """
+        Add activity to the participant.
+
+        This method checks if the current activity is
+        already peformed by the user, if yes, then it
+        increase the 'number_of_times' field with one with
+        creating a new activity as the performing time for the
+        second activity would be different.
+        If not then it adds a new activity to the participant.
+        """
+        if self.activities.filter(name=activity_string).exists():
+            # Number of times this activity has been already performed
+            previously_performed = Activity.objects.filter(
+                name=activity_string,
+                performer=self.username).count()
+
+            # Creating a new activity with increasing the number_of_times
+            # With one because the performed_at and updated_at time would
+            # be different for this activity
+            new_activity = Activity.objects.create(
+                name=activity_string, points=points,
+                number_of_times=previously_performed + 1,
+                performed_at=performed_at,
+                updated_at=updated_at,
+                performer=self)
+            self.activities.add(new_activity)
+        else:
+            activity = Activity.objects.create(name=activity_string,
```

Comment:
args are aligned differently here than in the other branch of the `if`.

that makes the reader not see that they are very similar sets of code.